### DISCUSSION TITLE
How do I find the power of a number?

### DISCUSSION THREAD

**Discussion by dyc3** [sympy/sympy#23279] [Q&A] [answered]
Suppose I have the number `x = 1000`. Does sympy have a method to determine that this is `10**3`, and the exponent is `3`?

Other examples:
`23000` -> `2.3*10**4` -> `4`
`730020` -> `7.3002*10**5` -> `5`

**Comment by kevinwm0** [comment, marked as answer]
You would need to provide the base. The log function will do this. 1000 = 10^3, but also 1000 ~= 7^3.55

If you're only interested in the power of 10 (that is - the highest power of ten less than the number), then floor(log(x)) will do it.
